pkg/liquidity-mining/test/ChildChainGauge.test.ts

```diff
@@ -0,0 +1,414 @@
+        // users.
+        const expectedBalanceChanges = [
+          {
+            account: gauge.address,
+            changes: selectedRewards.reduce((acc, token) => {
+              acc[token.symbol] = ['near', rewardAmount.mul(claimerStake).div(gaugeTotalSupply).mul(-1)]; // Outgoing
+              return acc;
+            }, {} as Record<string, Comparison>),
+          },
+          {
+            account: claimer,
+            changes: selectedRewards.reduce((acc, token) => {
+              acc[token.symbol] = ['near', rewardAmount.mul(claimerStake).div(gaugeTotalSupply)];
+              return acc;
+            }, {} as Record<string, Comparison>),
+          },
+        ];
+
+        // All the rewards are available in the gauge at this point.
+        // We claim only the selected ones by sending the indices that correspond to them, but we monitor the balance
+        // change of all the rewards inconditionally. This way, we ensure that only the selected rewards are transferred
+        // and the remaining ones are not affected.
+        await expectBalanceChange(claimFunction, rewards, expectedBalanceChanges);
+      });
+    }
+
+    function itClaimsRewards(user1Stake: BigNumber, user2Stake: BigNumber) {
+      sharedBeforeEach('stake BPT and wait', async () => {
+        await stakeBPT(user1Stake, user2Stake);
+        // Rewards are distributed throughout a week
+        await advanceToTimestamp((await currentTimestamp()).add(WEEK));
+      });
+
+      context('all rewards with no explicit indexes', () => {
+        sharedBeforeEach(() => {
+          selectedRewards = rewards;
+        });
+
+        itTransfersRewardsToClaimer(claimAll);
+      });
+
+      context('all rewards using indexes', () => {
+        sharedBeforeEach(() => {
+          selectedRewards = rewards;
+        });
+
+        itTransfersRewardsToClaimer(claimSelected);
+      });
+
+      context('selecting consecutive rewards', () => {
+        sharedBeforeEach(() => {
+          selectedRewards = rewards.subset(3, 2);
+        });
+
+        itTransfersRewardsToClaimer(claimSelected);
+      });
+
+      context('selecting random rewards', () => {
+        sharedBeforeEach(() => {
+          selectedRewards = new TokenList([rewards.get(4), rewards.get(7), rewards.get(1)]);
+        });
+
+        itTransfersRewardsToClaimer(claimSelected);
+      });
+    }
+
+    sharedBeforeEach('grant add_reward permission to admin', async () => {
+      const action = await actionId(vault.authorizerAdaptorEntrypoint, 'add_reward', gauge.interface);
+      await vault.grantPermissionGlobally(action, admin);
+    });
+
+    sharedBeforeEach('add reward', async () => {
+      await Promise.all(
+        rewards.addresses.map((reward) =>
+          vault.authorizerAdaptorEntrypoint
+            .connect(admin)
+            .performAction(gauge.address, gauge.interface.encodeFunctionData('add_reward', [reward, admin.address]))
+        )
+      );
+    });
```

Comment:
It'd be good to at least test the scenario where there are no reward tokens, and a scenario in which there are _some_ reward tokens, where otherwise valid indices (e.g. 5) then fail